tests/testsuite/doc.rs

```diff
@@ -1168,6 +1168,39 @@ fn doc_workspace_open_help_message() {
         .run();
 }
 
+#[cargo_test]
+#[cfg(not(windows))] // `echo` may not be available
+fn doc_extern_map_local() {
+    assert!(is_nightly());
+
+    let p = project()
+        .file(
+            "Cargo.toml",
+            r#"
+                [package]
+                name = "foo"
+                version = "0.1.0"
+            "#,
+        )
+        .file("src/lib.rs", "")
+        .build();
+
+    p.change_file(
+        ".cargo/config.toml",
+        r#"
+[doc.extern-map]
+std = "local"
+    "#,
+    );
+
+    p.cargo("doc -v --no-deps -Zrustdoc-map --open")
+        .env("BROWSER", "echo")
+        .masquerade_as_nightly_cargo()
+        .with_stderr_contains("[..] Documenting foo v0.1.0 ([..])")
+        .with_stderr_does_not_contain("warning: unused config key `doc.extern-map` [..]")
```

Comment:
`with_stderr_does_not_contain` can be dangerous to use, because there is no way to tell if it continues to work in the future (say, for example, the warning text was slightly changed).  Can this instead use a `with_stderr()` to just check all the output?  It would look something like [this](https://github.com/rust-lang/cargo/blob/1fc64069fd3b24c506b607fdc3404cf87014fa1a/tests/testsuite/doc.rs#L1659-L1667) (plus the "Opening" line).  That should help ensure that no warnings are emitted at all, and that nothing unexpected is displayed.